CMakeLists.txt

```diff
@@ -1123,6 +1123,14 @@ if (EXISTS "${HDF5_SOURCE_DIR}/fortran" AND IS_DIRECTORY "${HDF5_SOURCE_DIR}/for
         set (HDF5_INSTALL_MOD_FORTRAN "SHARED")
       endif ()
     endif ()
+    if (NOT DEFINED HDF5_INSTALL_FMOD_DIR)
+      set (HDF5_INSTALL_FMOD_DIR ${HDF5_INSTALL_INCLUDE_DIR})
+      set (HDF5_INSTALL_SHARED_FMOD_DIR ${HDF5_INSTALL_INCLUDE_DIR}/shared)
+      set (HDF5_INSTALL_STATIC_FMOD_DIR ${HDF5_INSTALL_INCLUDE_DIR}/static)
+    else ()
+      set (HDF5_INSTALL_SHARED_FMOD_DIR ${HDF5_INSTALL_FMOD_DIR})
```

Comment:
I have an issue with this being the same folder if both static and shared is being built. One set will overwrite the other.
While this may be okay for the majority of the platform-compiler pairs - the mod files are still objects and are not always equivalent. 
I think it depends on HDF5_INSTALL_MOD_FORTRAN not being "no" - then this is okay if it is "no" then something needs to be done about one or the other. Maybe send static to a subdir and leave shared to the DIR?